pkg/endpoints/pipeline_test.go

```diff
@@ -26,216 +26,262 @@ import (
 		},
 	}
+	var allTasks int
 
-	var tasks []v1alpha1.Task
-	tasks = append(tasks, ns1Tasks...)
-	tasks = append(tasks, ns2Tasks...)
-
-	for _, task := range tasks {
-		_, err := r.PipelineClient.TektonV1alpha1().Tasks(task.ObjectMeta.Namespace).Create(&task)
-		if err != nil {
-			t.Fatalf("Error creating task '%s': %v\n", task.ObjectMeta.Name, err)
-		}
-	}
-
-	// Test getAllTasks function
+	// Test getAllTasks: no tasks
 	httpReq := dummyHTTPRequest("GET", fmt.Sprintf("%s/v1/namespaces/*/tasks", server.URL), nil)
-	response, _ := http.DefaultClient.Do(httpReq)
+	response, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		t.Fatalf("getAllTasks response error: %v\n", err)
+	}
 	responseTaskList := v1alpha1.TaskList{}
 	if err := json.NewDecoder(response.Body).Decode(&responseTaskList); err != nil {
 		t.Fatalf("Error decoding getAllTasks response: %v\n", err)
 	} else {
-		if len(responseTaskList.Items) != len(tasks) {
-			t.Errorf("All expected tasks were not returned: expected %v, actual %v", len(tasks), len(responseTaskList.Items))
+		if len(responseTaskList.Items) != 0 {
+			t.Fatal("No tasks expected")
 		}
 	}
-
-	// Test getTask function
-	for _, task := range tasks {
-		httpReq := dummyHTTPRequest("GET", fmt.Sprintf("%s/v1/namespaces/%s/tasks/%s", server.URL, task.ObjectMeta.Namespace, task.Name), nil)
-		response, err := http.DefaultClient.Do(httpReq)
-		if err != nil {
-			t.Fatalf("Error getting task '%s': %v\n", task.Name, err)
-			continue
-		}
-		responseTask := v1alpha1.Task{}
-		if err := json.NewDecoder(response.Body).Decode(&responseTask); err != nil {
-			t.Fatalf("Error decoding getTask response: %v\n", err)
-			continue
-		}
-		if !reflect.DeepEqual(responseTask, task) {
-			t.Fatalf("Response object %v did not equal expected %v\n", responseTask, task)
+	// Create tasks
+	for _, namespacedTasks := range namespaceTaskMap {
+		for _, task := range namespacedTasks {
+			_, err := r.PipelineClient.TektonV1alpha1().Tasks(task.ObjectMeta.Namespace).Create(&task)
+			if err != nil {
+				t.Fatalf("Error creating task '%s': %v\n", task.ObjectMeta.Name, err)
+			}
+			allTasks++
 		}
 	}
-
-	// Test getAllTasks for namespace function
-	httpReq = dummyHTTPRequest("GET", fmt.Sprintf("%s/v1/namespaces/%s/tasks", server.URL, namespace), nil)
+	// Test getAllTasks: all namespaces
+	httpReq = dummyHTTPRequest("GET", fmt.Sprintf("%s/v1/namespaces/*/tasks", server.URL), nil)
 	response, err = http.DefaultClient.Do(httpReq)
 	if err != nil {
 		t.Fatalf("getAllTasks response error: %v\n", err)
+	}
+	responseTaskList = v1alpha1.TaskList{}
+	if err := json.NewDecoder(response.Body).Decode(&responseTaskList); err != nil {
+		t.Fatalf("Error decoding getAllTasks response: %v\n", err)
 	} else {
-		responseTaskList := v1alpha1.TaskList{}
-		if err := json.NewDecoder(response.Body).Decode(&responseTaskList); err != nil {
-			t.Fatalf("Error decoding getAllTasks response: %v\n", err)
+		if len(responseTaskList.Items) != allTasks {
```

Comment:
Instead of just checking that the number of tasks matches, please iterate over the list and check that each of the tasks has the correct content (you'll probably want a helper function for this that can be reused in this test file).